compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/FirCallCompletionResultsWriterTransformer.kt

```diff
@@ -436,7 +437,29 @@ class FirCallCompletionResultsWriterTransformer(
         // Control flow info is necessary prerequisite because we collect return expressions in that function
         //
         // Example: second lambda in the call like list.filter({}, {})
-        if (!dataFlowAnalyzer.isThereControlFlowInfoForAnonymousFunction(anonymousFunction)) return anonymousFunction.compose()
+        if (!dataFlowAnalyzer.isThereControlFlowInfoForAnonymousFunction(anonymousFunction)) {
```

Comment:
This part is probably acceptable but I'd like to have a second opinion here @demiurg906

Reply:
I think it's fine, but it's better to extract this code to method (`transformAnonymousFunction` already is big enough)

Reply:
Extracted as suggested.